edith-step/src/App.js

```diff
@@ -9,38 +9,49 @@ import axios from 'axios';
 class App extends Component {
   constructor() {
     super();
-    this.state = { "chartType" : LineChart }; 
-    this.updateChartType = this.updateChartType.bind(this);   
+    this.state = { "chartType" : LineChart, "dateSelection": ""}; 
+    this.updateChartType = this.updateChartType.bind(this); 
+    this.showWeeklyChart = this.showWeeklyChart.bind(this);  
+    this.revertChart = this.revertChart.bind(this);
   }
 
   updateChartType(event) {
     switch(event.target.value) {
-      case "LineChart": 
+      case "line":
         this.setState({"chartType": LineChart});
         break;
-      case "BarChart":
-        this.setState({"chartType": BarChart});
+      case "bar":
+        this.setState({"chartType": BarGraph});
+        break;
+      case "doughnut":
+        this.setState({"chartType": DoughnutChart});
         break;
     }
   }
 
+  showWeeklyChart(dateSelection) {
+    this.setState({"chartType":  DoughnutChart, "dateSelection": dateSelection});
+  }
+  
+  revertChart() {
```

Comment:
nit: these functions are clear, but javadocs could be useful on each of them for the linter

Reply:
Okay, I added them.